x/feemarket/post/fee.go

```diff
@@ -73,29 +76,32 @@ func (dfd FeeMarketDeductDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simul
 	}
 
 	baseFee := sdk.NewCoin(params.FeeDenom, state.BaseFee)
-	minGasPrices := sdk.NewCoins(baseFee)
 
-	fee := feeTx.GetFee()
+	feeCoins := feeTx.GetFee()
 	gas := ctx.GasMeter().GasConsumed() // use context gas consumed
 
+	if len(feeCoins) > 1 {
+		return ctx, feemarkettypes.ErrTooManyFeeCoins
+	}
+
 	ctx.Logger().Info("fee deduct post handle",
-		"min gas prices", minGasPrices,
+		"min gas prices", baseFee,
 		"gas consumed", gas,
 	)
 
 	if !simulate {
-		fee, tip, err = ante.CheckTxFees(ctx, minGasPrices, feeTx, false)
+		feeCoin, tip, err = ante.CheckTxFee(ctx, baseFee, feeTx, false, dfd.feemarketKeeper.GetDenomResolver())
 		if err != nil {
 			return ctx, err
 		}
 	}
 
 	ctx.Logger().Info("fee deduct post handle",
-		"fee", fee,
+		"fee", feeCoins,
 		"tip", tip,
 	)
 
-	if err := dfd.DeductFeeAndTip(ctx, tx, fee, tip); err != nil {
+	if err := dfd.DeductFeeAndTip(ctx, tx, feeCoin, tip); err != nil {
```

Comment:
this is deducting the converted fee or the fees that the person set on the tx?

Reply:
im assuming we want to deduct the fees set on the tx not the converted amount since they might not have that in their account.